Tensile/TensileCreateLibrary.py

```diff
@@ -518,53 +523,45 @@ def buildKernelSourceAndHeaderFiles(results, outputPath):
-                os.path.join(os.path.normcase(outputPath), "Kernels" + kernelSuffix)
-            ].append((err, src, header, kernelName))
-        else:
-            filesToWrite[os.path.join(os.path.normcase(outputPath), kernelName)].append(
-                (err, src, header, kernelName)
-            )
+            key = pathJoin(filename)
+        elif mergeFiles:
+            kernelSuffix = str(validKernelCount % numMergedFiles) if numMergedFiles > 1 else ""
+            key = pathJoin(f"Kernels{kernelSuffix}")
+
+        filesToWrite[key].append((err, src, header, kernelName))
         validKernelCount += 1
 
     # Ensure there's at least one kernel file for helper kernels
-    if globalParameters["LazyLibraryLoading"] or (
-        globalParameters["MergeFiles"] and not kernelsToWrite
-    ):
-        kernelSuffix = ""
-        if globalParameters["NumMergedFiles"] > 1:
-            kernelSuffix = "0"
-
-        filesToWrite[os.path.join(os.path.normcase(outputPath), "Kernels" + kernelSuffix)] = []
+    if lazyLoading or (mergeFiles and not kernelsToWrite):
+        kernelSuffix = "0" if numMergedFiles > 1 else ""
+        filesToWrite[pathJoin(f"Kernels{kernelSuffix}")] = []
+
+    def writeHeader(hdrFile):
+        hdrFile.write(CHeader)
+        hdrFile.write("#pragma once\n")
+        if globalParameters["RuntimeLanguage"] == "HIP":
+            hdrFile.write("#include <hip/hip_runtime.h>\n")
+            hdrFile.write("#include <hip/hip_ext.h>\n\n")
+        hdrFile.write('#include "KernelHeader.h"\n\n')
+
+    def writeSource(srcFile, filename):
+        srcFile.write(CHeader)
+        srcFile.write(f'#include "{filename}.h"\n')
 
     # Write kernel data to files
     # Parse list of files and write kernels
     for filename, kernelList in filesToWrite.items():
-        with open(filename + ".h", "w", encoding="utf-8") as kernelHeaderFile, open(
-            filename + ".cpp", "w", encoding="utf-8"
-        ) as kernelSourceFile:
-
-            kernelSourceFile.write(CHeader)
-            kernelHeaderFile.write(CHeader)
-            kernelSourceFile.write('#include "{}.h"\n'.format(filename))
-            kernelHeaderFile.write("#pragma once\n")
-            if globalParameters["RuntimeLanguage"] == "HIP":
-                kernelHeaderFile.write("#include <hip/hip_runtime.h>\n")
-                kernelHeaderFile.write("#include <hip/hip_ext.h>\n\n")
-            kernelHeaderFile.write('#include "KernelHeader.h"\n\n')
-
-            for err, src, header, kernelName in kernelList:
-                kernelSourceFile.write(src)
-                kernelHeaderFile.write(header)
+        with open(f"{filename}.h", "w", encoding="utf-8") as hdrFile, open(
+            f"{filename}.cpp", "w", encoding="utf-8"
+        ) as srcFile:
+
+            writeHeader(hdrFile)
+            writeSource(srcFile, filename)
+            for _, src, header, _ in kernelList:
+                srcFile.write(src)
+                hdrFile.write(header)
```

Comment:
Function?

Reply:
The PR description has been updated with details about the new function.